.github/workflows/ids-check.yml

```diff
@@ -0,0 +1,75 @@
+name: ids-check
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    if: github.repository_owner == 'llvm'
+    name: ids-check
+    runs-on: ubuntu-24.04
+
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      with:
+        repository: compnerd/ids
+        path: ${{ github.workspace }}/ids
+        fetch-depth: 0
+
+    - uses: actions/checkout@v4
+      with:
+        path: ${{ github.workspace }}/llvm-project
+        fetch-depth: 2
+
+    - name: install dependencies
+      run: |
+        sudo apt install -y clang-18 ninja-build libclang-18-dev
+        pip install lit
+
+    - name: Configure and build minimal LLVM for use by ids
+      run: |
+        cmake -B ${{ github.workspace }}/llvm-project/build/ \
+              -D CMAKE_BUILD_TYPE=Release \
+              -S ${{ github.workspace }}/llvm-project/llvm/ \
+              -D LLVM_ENABLE_PROJECTS=clang \
+              -D LLVM_TARGETS_TO_BUILD="host" \
+              -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+              -G Ninja
+        cd ${{ github.workspace }}/llvm-project/build/
+        ninja -t targets all | grep "CommonTableGen: phony$" | grep -v "/" | sed 's/:.*//'
+
+    - name: Configure ids
+      run: |
+        cmake -B ${{ github.workspace }}/ids/build/ \
+              -S ${{ github.workspace }}/ids/ \
+               -D CMAKE_BUILD_TYPE=Release \
+               -D CMAKE_C_COMPILER=clang \
+               -D CMAKE_CXX_COMPILER=clang++ \
+               -D CMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld \
+               -D LLVM_DIR=/usr/lib/llvm-18/lib/cmake/llvm/ \
+               -D Clang_DIR=/usr/lib/llvm-18/lib/cmake/clang/ \
+               -D FILECHECK_EXECUTABLE=$(which FileCheck-18) \
+               -D LIT_EXECUTABLE=$(which lit)
+
+    - name: Build ids
+      run: |
+        cmake --build ${{ github.workspace }}/ids/build \
+              --config Release \
+              --parallel $(nproc --all)
+
+    - name: Run ids over compilation database
+      run: |
+        cd ${{ github.workspace }}/llvm-project
+        export H_OR_CPP_FILES_CHANGED_LAST_COMMIT=$(git diff --name-only HEAD~1 HEAD -- '*.cpp' '*.h')
```

Comment:
Does this assume pull requests to have just one commit?

Reply:
The ref that gets cloned from the `pull_request` event in Github contains the PR merged into main. So if you diff the last commit (the merge commit), you get the diff between main and the PR.

Reply:
Ok, thanks for the clarification!